### ISSUE TITLE
Step link needs an href attribute so that it's keyboard navigable

### ISSUE THREAD

**Issue by mmazanec22** [oruga-ui/oruga#455]
### Overview of the problem

**Oruga** version: 0.5.9
**Vuejs** version: 3.2.41
**OS/Browser**: Pop!_OS/Chrome

### Description

The step nav bar is not currently operable via keyboard because the step `a` tags to not have `href` attributes.

### Steps to reproduce

Set the steps to be clickable, and then try to use the tab key to navigate through the page.

### Expected behavior

I would expect the clickable step `nav` links to have `href` attributes so that it's keyboard operable, and to be able to set those `href` attributes. I would also expect the `nav` to just be a regular `ol` with some aria markup and not use semantic `nav` and link elements when the step progress indicator is not interactive.

### Actual behavior

The links are clickable when the steps are set to clickable, but the links cannot receive keyboard focus. A step nav with no clickable elements is a semantic nav element with links despite not being interactive at all, which is potentially misleading to screen reader users.

(Edited because I mixed up animated and clickable.)

**Comment by jtommy** [comment]
Can you make a PR?

**Comment by mlmoravek** [comment]
Maybe its easier to add tabindex="0" and role="button" as needed, to make them accessible by keyboard?
